src/array.rs

```diff
@@ -1480,6 +1480,21 @@ unsafe fn clone_elements<T: Element>(elems: &[T], data_ptr: &mut *mut T) {
     }
 }
 
+/// Implementation of functionality for [`PyArray<T, D>`].
+#[doc(alias = "PyArray")]
+// TODO: seal this trait
```

Comment:
Could also seal and re-export this trait even if incomplete? Thanks!

Reply:
I have used `PyUntypedArrayMethods` as a super trait here, because it allows for more default implementations of `PyArrayMethods` in the future. Since `PyUntypedArrayMethods` is sealed, this also seals `PyArrayMethods`. (And it also makes semantic sense, every `PyArray` is also a `PyUntypedArray`)